src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/EiffelSourceChangeCreatedEventGenerator.java

```diff
@@ -0,0 +1,66 @@
+/*
+   Copyright 2019 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.gerrit.plugins.eiffel.events.generators;
+
+import com.ericsson.gerrit.plugins.eiffel.configuration.EiffelPluginConfiguration;
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelSourceChangeCreatedEvent;
+import com.google.gerrit.server.events.PatchSetCreatedEvent;
+
+public class EiffelSourceChangeCreatedEventGenerator extends EiffelEventGenerator {
+    private static final String TYPE = "EiffelSourceChangeCreatedEvent";
+    private static final String VERSION = "4.0.0";
+
+    /**
+     * Extracts information from the PatchSetCreatedEvent and generates an
+     * EiffelSourceChangeCreatedEvent
+     *
+     * @param patchSetCreatedEvent
+     * @param pluginConfig
+     * @return EiffelSourceChangeCreatedEvent
+     */
+    public static EiffelSourceChangeCreatedEvent generate(PatchSetCreatedEvent patchSetCreatedEvent,
+            EiffelPluginConfiguration pluginConfig) {
+        final String projectName = patchSetCreatedEvent.change.get().project;
```

Comment:
It is a bit hard to figure out what `get()` does. Is there any documentation how these events are build up or did you just look at the code?

Reply:
Yes it's a bit wierd way. We tried following documentations but the API for 2.14 seems to be a bit other than the documentation.

Reply:
Is it possible to give a link to the documentation somewhere or does that go against your principle?

Reply:
We feel that external libs shouldn't be documented in our javadoc.

Reply:
After reading this again, could you at least break out the `patchSetCreatedEvent.change.get()` to a variable explaining what it is doing?

Reply:
Well it gets the ChangeAttribute object that contains the actual information fields. I could not find documentation for this and there is no javadocs. I only know what type of object is but I don't know why it's done this way. Would simply ChangeAttribute changeAttribute = patchSetCreatedEvent.change.get(); be enough?

Reply:
I talked with @e-edling-ericsson and we simply break them out as separate variables now.